Frameworks/UIKit/UIImage.mm

```diff
@@ -44,487 +44,416 @@ @interface UIImage ()
+    _scale = scale;
+    _orientation = orientation;
+  }
 
-    return SUCCEEDED(hr);
+  return self;
 }
 
-static bool loadGIF(UIImage* dest, void* bytes, int length) {
-    return loadImageWithWICDecoder(dest, GUID_ContainerFormatGif, bytes, length);
-}
+- (instancetype)_initWithCopyOfImage:(UIImage *)imageToCopy
+                   WithRenderingMode:(UIImageRenderingMode)renderingMode {
 
-static bool loadBMP(UIImage* dest, void* bytes, size_t length) {
-    return loadImageWithWICDecoder(dest, GUID_ContainerFormatBmp, bytes, length);
-}
+  RETURN_NULL_IF(!imageToCopy);
 
-static bool loadTIFF(UIImage* dest, void* bytes, int length) {
-    return loadImageWithWICDecoder(dest, GUID_ContainerFormatTiff, bytes, length);
+  if (self = [self initWithCGImage:getImage(imageToCopy)]) {
+    _scale = imageToCopy->_scale;
+    _orientation = imageToCopy->_orientation;
+    _imageInsets = imageToCopy->_imageInsets;
+    _imageStretch = imageToCopy->_imageStretch;
+    _renderingMode = renderingMode;
+  }
+  return self;
 }
 
 /**
  @Status Interoperable
 */
-- (instancetype)initWithContentsOfFile:(NSString*)pathAddr {
-    if (pathAddr == nil) {
-        return nil;
-    }
+- (instancetype)initWithContentsOfFile:(NSString *)pathAddr {
+  RETURN_NULL_IF(!pathAddr);
 
-    _scale = 1.0f;
-    _imageStretch.origin.x = 0.0f;
-    _imageStretch.origin.y = 0.0f;
-    _imageStretch.size.width = 1.0f;
-    _imageStretch.size.height = 1.0f;
+  if (self = [self init]) {
 
-    NSBundle* bundle = [NSBundle mainBundle];
+    NSBundle *bundle = [NSBundle mainBundle];
 
-    const char* path = (char*)[pathAddr UTF8String];
+    const char *path = (char *)[pathAddr UTF8String];
     bool found = false;
-    char* pathStr = NULL;
+    char *pathStr = NULL;
 
     if (strlen(path) == 0) {
-        TraceVerbose(TAG, L"UIImage: path is blank");
-        return nil;
+      TraceVerbose(TAG, L"UIImage: path is blank");
+      [self release];
+      return nil;
     }
 
     if (strrchr(path, '.') != NULL && GetCACompositor()->screenScale() > 1.5f) {
-        size_t newStrSize = strlen(path) + 10;
-        char* newStr = (char*)IwMalloc(newStrSize);
-        const char* pathEnd = strrchr(path, '.');
-        memcpy(newStr, path, pathEnd - path);
-        newStr[pathEnd - path] = 0;
-        strcat_s(newStr, newStrSize, "@2x");
-        strcat_s(newStr, newStrSize, pathEnd);
-
-        pathStr = IwStrDup(newStr);
-
-        if (EbrAccess(pathStr, 0) == -1) {
-            id pathFind =
-                [bundle pathForResource:[NSString stringWithCString:newStr] ofType:nil inDirectory:nil forLocalization:@"English"];
-
-            if (pathFind != nil) {
-                path = (char*)[pathFind UTF8String];
-                if (pathStr)
-                    IwFree(pathStr);
-                pathStr = IwStrDup(path);
-                found = true;
-            }
-        } else {
-            found = true;
+      size_t newStrSize = strlen(path) + 10;
```

Comment:
this is getting cleaned up by brain. Leaving it as is for now.

---
In reply to: [84145944](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84145944) [](ancestors = 84145944)